mlflow/utils/autologging_utils.py

```diff
@@ -13,7 +13,41 @@ def try_mlflow_log(fn, *args, **kwargs):
         warnings.warn("Logging to MLflow failed: " + str(e), stacklevel=2)
 
 
-def param_logger(fn, args=[], kwargs=None, unlogged=[]):
+def get_unspecified_default_args(user_args, user_kwargs, all_param_names, all_default_values):
+    """
+    Determine which default values are used in a call, given args and kwargs that are passed in.
+
+    :param user_args: list of arguments passed in by the user
+    :param user_kwargs: dictionary of kwargs passed in by the user
+    :param all_param_names: names of all of the parameters of the function
+    :param all_default_values: values of all default parameters
+    :return: a dictionary mapping arguments not specified by the user -> default value
+    """
+    num_default_args = len(all_default_values)
+
+    # all_default_values correspond to the last len(all_default_values) elements of the arguments
+    default_param_names = all_param_names[-num_default_args:]
```

Comment:
BTW, this logic seems to assume that `num_default_args > 0` (if it's 0 then `all_param_names[0:]` returns the whole list), probably better to compute `num_args_without_default_value` earlier & write this as `all_param_names[num_args_without_default_value:]`

Reply:
Ah good catch, will fix